File: src/main/java/org/gabi/projet_gl/model/Task.java
package org.gabi.projet_gl.model;
import jakarta.persistence.*;
import org.antlr.v4.runtime.misc.NotNull;


@Entity
public class Task {
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long id;
  private String title;
  private String description;
  private boolean done;
  @ManyToOne
  @JoinColumn(name = "user_id")
  private AppUser resp;
  @ManyToOne
  @JoinColumn(name = "project_id")
  private Project project;
  public Long getId() {
    return id;
  }
  public void setId(Long id) {
    this.id = id;
  }
  public String getTitle() {
    return title;
  }
  public void setTitle(String title) {
    this.title = title;
  }
  public String getDescription() {
    return description;
  }
  public void setDescription(String description) {
    this.description = description;
  }
  public boolean isDone() {
    return done;
  }
  public void setDone(boolean done) {
    this.done = done;
  }
  public AppUser getResp() {
    return resp;
  }
  public void setResp(AppUser resp) {
    this.resp = resp;
  }
  public Project getProject() {
    return project;
  }
  public void setProject(Project project) {
    this.project = project;
  }

  public Task( String title, String description, boolean done, AppUser usr, Project pj) {
    this.title = title;
    this.description = description;
    this.done = done;
    this.resp = usr;
    this.project = pj;
  }
  public Task(){};
}

File: frontend/src/app/task-class/task/task.ts
import {Project} from '../../project-class/project/project';

export class Task{

  private _id : string = "";
  private _title : string = "";
  private _description : string = "";
  private _project : Project = new Project();
  private _user_id : string = "";
  private _done : boolean = false;

  constructor() {
  }

  get id() : string{
    return this._id;
  }
  set id(task_id : string){
    this._id = task_id;
  }

  get title() : string{
    return this._title;
  }
  set title(title : string){
    this._title = title;
  }

  get description() : string{
    return this._description;
  }
  set description(description : string){
    this._description = description;
  }

  get done() : boolean{
    return this._done;
  }
  set done(done : boolean){
    this._done = done;
  }

  get project(): Project {
    return this._project;
  }

  set project(value: Project) {
    this._project = value;
  }


  get user_id(): string {
    return this._user_id;
  }

  set user_id(value: string) {
    this._user_id = value;
  }

}
